src/hotspot/cpu/riscv/macroAssembler_riscv.hpp

```diff
@@ -1397,7 +1397,13 @@ class MacroAssembler: public Assembler {
 private:
   void cmp_x2i(Register dst, Register src1, Register src2, Register tmp, bool is_signed = true);
 
+  void compress_bits(Register dst, Register src, Register mask, Register tmp, bool is_long);
+
 public:
+  // compress bits, i.e. j.l.Long::compress.
```

Comment:
All `compress_bits` methods are only used for C2, maybe we could move them into `C2_MacroAssembler`.

Reply:
Yes, it makes sense.